services/communitytokens/api.go

```diff
@@ -679,3 +868,91 @@ func (api *API) estimateMethod(ctx context.Context, chainID uint64, contractAddr
 	}
 	return estimate + uint64(float32(estimate)*0.1), nil
 }
+
+// Gets signer public key from smart contract with a given chainId and address
+func (api *API) GetSignerPubKey(ctx context.Context, chainID uint64, contractAddress string) (string, error) {
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	contractInst, err := api.NewOwnerTokenInstance(chainID, contractAddress)
+	if err != nil {
+		return "", err
+	}
+	signerPubKey, err := contractInst.SignerPublicKey(callOpts)
+	if err != nil {
+		return "", err
+	}
+	return common.Bytes2Hex(signerPubKey), nil
+}
+
+// Gets signer public key directly from deployer contract
+func (api *API) SafeGetSignerPubKey(ctx context.Context, chainID uint64, communityID string) (string, error) {
```

Comment:
Why is it "safe"?

Reply:
There are 2 safe functions: SafeGetSignerPubKey and SafeGetOwnerTokenAddress. I called them "safe" because I wanted to indicate that that these functions use directly deployer contract so we can trust this information.
If it is not clear enough I can change it of course. 
Another way is to get owner token address from, eg. CommunityDescription and use GetSignerPubKey(chainID, contractAddress) if we trust that the contract address in CommunityDescription was not malformed.

Reply:
Is there a scenario where clients of the public API would be interested in the "non-safe" version? If not, what do you think about renaming the methods as follows:
```
func (api *API) getSignerPubKey(ctx context.Context, chainID uint64, contractAddress string) (string, error) {...}
func (api *API) GetSignerPubKey(ctx context.Context, chainID uint64, communityID string) (string, error) {...}
```

Reply:
Honestly, I am not sure. I created the API before the logic.
However, I am open to any changes. At this moment I am not sure which function will be used.